agent/engine/serviceconnect/manager_linux.go

```diff
@@ -372,17 +370,40 @@ func (m *manager) AugmentInstanceContainer(task *apitask.Task, container *apicon
 	return nil
 }
 
-// LoadImage helps load the AppNetAgent container image for the agent
+func (agent *manager) setLoadedAppnetVerion(appnetInterfaceVersion string) {
+	agent.AppnetInterfaceVersion = appnetInterfaceVersion
+}
+
+// LoadImage helps load the AppNetAgent container image for the agent latest supported
+// AppNet interface version by looking for the AppNet agent tar name from supported list
+// of AppNet versions from highest to lowest version when loading AppNet image
 func (agent *manager) LoadImage(ctx context.Context, _ *config.Config, dockerClient dockerapi.DockerClient) (*types.ImageInspect, error) {
-	logger.Debug("Loading appnet agent container tarball:", logger.Fields{
-		field.Image: agent.AgentContainerTarballPath,
-	})
-	if err := loader.LoadFromFile(ctx, agent.AgentContainerTarballPath, dockerClient); err != nil {
-		return nil, err
+	var err error
+	for _, supportedAppnetInterfaceVersion := range getSupportedAppnetInterfaceVersions() {
+		agentContainerTarballPath := fmt.Sprintf(defaultAgentContainerTarballPathFormat, supportedAppnetInterfaceVersion)
+		if _, err = os.Stat(agentContainerTarballPath); err != nil {
+			logger.Warn(fmt.Sprintf("Agent container tarball unavailable: %s", agentContainerTarballPath), logger.Fields{
```

Comment:
Not blocking. Could we add "Appnet" or "AppNet" before "Agent container tarball unavailable" to avoid mixing it up with ECS Agent or other Agents? And could we have a standard for using "Appnet", "appnet" or "AppNet" in all warning/error messages?  Thanks!

Reply:
Ack! 

Since all these callouts are non blocking, will address them in the upcoming PR related to AppNet reload(including all the below comments as well) :)
Thank you!